This patch is needed to improve the probability that the MySQL tests pass
in Red Hat's build environment.  The 32-bit and 64-bit variants of a single
architecture may be built on the same machine at the same time, so we have
to ensure that the tests use different port numbers on a 64-bit arch than
on a 32-bit, else they will interfere with each other.  Also increase a
timeout that has been observed to get violated under heavy load.


diff -Naur mysql-4.1.9.orig/mysql-test/mysql-test-run.sh mysql-4.1.9/mysql-test/mysql-test-run.sh
--- mysql-4.1.9.orig/mysql-test/mysql-test-run.sh	2005-01-11 17:06:03.000000000 -0500
+++ mysql-4.1.9/mysql-test/mysql-test-run.sh	2005-01-15 19:44:10.780699826 -0500
@@ -200,12 +200,24 @@
 
 MASTER_RUNNING=0
 MASTER1_RUNNING=0
-MASTER_MYPORT=9306
-MASTER_MYPORT1=9307
 SLAVE_RUNNING=0
-SLAVE_MYPORT=9308 # leave room for 2 masters for cluster tests
-MYSQL_MANAGER_PORT=9305 # needs to be out of the way of slaves
-NDBCLUSTER_PORT=9350
+# hack to allow 32- and 64-bit tests to run concurrently on same build machine
+case `uname -m` in
+  ppc64 | s390x | x86_64)
+    MASTER_MYPORT=9406
+    MASTER_MYPORT1=9407
+    SLAVE_MYPORT=9408 # leave room for 2 masters for cluster tests
+    MYSQL_MANAGER_PORT=9405 # needs to be out of the way of slaves
+    NDBCLUSTER_PORT=9450
+    ;;
+  *)
+    MASTER_MYPORT=9306
+    MASTER_MYPORT1=9307
+    SLAVE_MYPORT=9308 # leave room for 2 masters for cluster tests
+    MYSQL_MANAGER_PORT=9305 # needs to be out of the way of slaves
+    NDBCLUSTER_PORT=9350
+    ;;
+esac
 MYSQL_MANAGER_PW_FILE=$MYSQL_TEST_DIR/var/tmp/manager.pwd
 MYSQL_MANAGER_LOG=$MYSQL_TEST_DIR/var/log/manager.log
 MYSQL_MANAGER_USER=root
@@ -230,7 +242,7 @@
 SLEEP_TIME_FOR_FIRST_MASTER=400		# Enough time to create innodb tables
 SLEEP_TIME_FOR_SECOND_MASTER=400
 SLEEP_TIME_FOR_FIRST_SLAVE=400
-SLEEP_TIME_FOR_SECOND_SLAVE=30
+SLEEP_TIME_FOR_SECOND_SLAVE=60
 CHARACTER_SET=latin1
 DBUSER=""
 START_WAIT_TIMEOUT=10
